thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java

```diff
@@ -1,250 +1,445 @@
-      if (dimensionValueMap == null) {
-        dimensionValueMap = new TreeMap<>();
-        dimensionData.get(metricFunction).put(dimensionName, dimensionValueMap);
-      }
-      // one row for each group by
-      int rows = thirdEyeResponse.getNumRowsFor(metricFunction);
-      for (int i = 0; i < rows; i++) {
-        Map<String, String> row = thirdEyeResponse.getRow(metricFunction, i);
-        String dimensionValue = row.get(dimensionName);
-        if (!dimensionValueMap.containsKey(dimensionValue)) {
-          Map<String, Double> value = new TreeMap<>();
-          dimensionValueMap.put(dimensionValue, value);
-          dimensionValueMap.get(dimensionValue).put(BASELINE_VALUE, 0d);
-          dimensionValueMap.get(dimensionValue).put(CURRENT_VALUE, 0d);
+    Map<String, ThirdEyeResponseRow> baselineResponseMap;
+    Map<String, ThirdEyeResponseRow> currentResponseMap;
+    List<MetricFunction> metricFunctions = baselineResponse.getMetricFunctions();
+    int numMetrics = metricFunctions.size();
+
+    List<Row> rows = new ArrayList<>();
+    if (hasGroupByTime) {
+
+      int numTimeBuckets = baselineRanges.size();
+
+      if (hasGroupByDimensions) {
+        // group by time and dimension  //contributor
+        baselineResponseMap = createResponseMapByTimeAndDimension(baselineResponse);
+        currentResponseMap = createResponseMapByTimeAndDimension(currentResponse);
+
+        Map<Integer, List<Double>> baselineMetricSums = getMetricSumsByTime(baselineResponse);
+        Map<Integer, List<Double>> currentMetricSums = getMetricSumsByTime(currentResponse);
+
+        Set<String> timeDimensionValues = new HashSet<>();
+        timeDimensionValues.addAll(baselineResponseMap.keySet());
+        timeDimensionValues.addAll(currentResponseMap.keySet());
+        Set<String> dimensionValues = new HashSet<>();
+        for (String timeDimensionValue : timeDimensionValues) {
+          String[] tokens = timeDimensionValue.split(TIME_DIMENSION_JOINER_ESCAPED);
+          String dimensionValue = tokens.length < 2 ? "" : tokens[1];
+          dimensionValues.add(dimensionValue);
         }
-        Double metricValue = Double.parseDouble(row.get(metricFunction.toString()));
-        dimensionValueMap.get(dimensionValue).put(BASELINE_OR_CURRENT, metricValue);
-      }
-    }
-  }
 
-  public static Row parseAggregationOnlyResponse(TimeOnTimeComparisonRequest comparisonRequest,
-      Map<ThirdEyeRequest, ThirdEyeResponse> queryResultMap) {
-    Map<MetricFunction, Map<String, Double>> metricOnlyData =
-        TimeOnTimeResponseParser.processAggregateResponse(queryResultMap);
+        String dimensionName = baselineResponse.getGroupKeyColumns().get(1);
 
-    Row.Builder rowBuilder = new Row.Builder();
-    rowBuilder.setBaselineStart(comparisonRequest.getBaselineStart());
-    rowBuilder.setBaselineEnd(comparisonRequest.getBaselineEnd());
+        // other row
```

Comment:
can you explain the logic, how many passes are we doing over the data to compute OTHER category